### PR TITLE
feat(debug): scaffold transparent tasks.json and launch.json

### PR DESCRIPTION
**Pull request by kuojianlu**
- add local debug transparency feature flag
- scaffold transparent tasks.json and launch.json when feture flag enables
- use comment-json to handle json with comments

E2E TEST: N/A

### CODE REVIEW

File: packages/fx-core/src/plugins/solution/fx-solution/debug/util/tasksTransparency.ts [hunk 1]
```diff
@@ -0,0 +1,686 @@
...
+  if (includeFrontend) {
+    prerequisites.push("devCert");
+    comments.push("53000, // tab service port");
+  }
+  if (includeBackend) {
+    prerequisites.push("func", "dotnet");
+    comments.push("7071, // backend service port", "9229, // backend debug port");
+  }
+  if (includeFuncHostedBot && !includeBackend) {
+    prerequisites.push("func", "dotnet");
```
Comment (XiaofuHuang) [inline packages/fx-core/src/plugins/solution/fx-solution/debug/util/tasksTransparency.ts:302]: func hosted bot may not need `dotnet`

Comment (kuojianlu) [reply packages/fx-core/src/plugins/solution/fx-solution/debug/util/tasksTransparency.ts:302]: Good catch, thx.

File: packages/fx-core/src/component/debug.ts [hunk 2]
```diff
@@ -487,67 +494,115 @@ export async function generateLocalDebugSettingsCommon(
...
 
-        await updateJson(
-          `${inputs.projectPath}/.vscode/tasks.json`,
-          {
-            version: "2.0.0",
-            tasks: tasks,
-          },
-          TasksNext.mergeTasks
-        );
+        if (await useTransparentTasks(inputs.projectPath)) {
```
Comment (swatDong) [inline packages/fx-core/src/component/debug.ts:504]: Just for more readable code (avoid soo many conditions...). Consider creating `generator` arrays with all possible functions.
Anyway, take this as further improvements.

``` typescript
private readonly taskGenerators = ...; // <version, <isM365, function>>
...
{
    const version = ...; // legacy, next, transparent
    const isM365 = ...; // true | false
    const tasks = await taskGenerators[version][isM365](args...);
}
...
```

**Review by swatDong — DISMISSED** [review]

File: packages/fx-core/src/plugins/solution/fx-solution/debug/util/tasksTransparency.ts [hunk 3]
```diff
@@ -0,0 +1,686 @@
...
+  };
+  if (includeFrontend) {
+    result.args.projects.push({
+      cwd: "${workspaceFolder}/tabs",
+      npmInstallArgs: ["--no-audit"],
+    });
+  }
+  if (includeBackend) {
+    result.args.projects.push({
+      cwd: "${workspaceFolder}/bot",
```
Comment (XiaofuHuang) [inline packages/fx-core/src/plugins/solution/fx-solution/debug/util/tasksTransparency.ts:348]: May be "${workspaceFolder}/api"?

File: packages/fx-core/src/plugins/solution/fx-solution/debug/util/tasksTransparency.ts [hunk 4]
```diff
@@ -0,0 +1,686 @@
...
+}
+
+function startLocalTunnel(): Record<string, unknown> {
+  return {
+    label: "Start local tunnel",
+    type: "teamsfx",
+    command: "debug-start-local-tunnel",
+    args: {
+      configFile: ".fx/configs/ngrok.yml",
+      useGlobalNgrok: false,
```
Comment (XiaofuHuang) [inline packages/fx-core/src/plugins/solution/fx-solution/debug/util/tasksTransparency.ts:383]: TODO(xiaofhua): add `tunnelName` into the config file

File: packages/fx-core/src/plugins/solution/fx-solution/debug/util/tasksTransparency.ts [hunk 5]
```diff
@@ -0,0 +1,686 @@
...
+    comments.push("7071, // backend service port", "9229, // backend debug port");
+  }
+  if (includeFuncHostedBot && !includeBackend) {
+    prerequisites.push("func", "dotnet");
+  }
+  if (includeBot) {
+    prerequisites.push("ngrok");
+    comments.push("3978, // bot service port", "9239, // bot debug port");
+  }
+  prerequisites.push("ports");
```
Comment (XiaofuHuang) [inline packages/fx-core/src/plugins/solution/fx-solution/debug/util/tasksTransparency.ts:308]: TODO(xiaofhua): `ports` -> `portsOccupation`

File: packages/fx-core/src/plugins/solution/fx-solution/debug/util/tasksTransparency.ts (hunk 3 above)
Comment (kuojianlu) [reply packages/fx-core/src/plugins/solution/fx-solution/debug/util/tasksTransparency.ts:348]: Good catch.

**Review by XiaofuHuang — DISMISSED** [review]

File: packages/fx-core/src/component/debug.ts (hunk 2 above)
Comment (kuojianlu) [reply packages/fx-core/src/component/debug.ts:498]: Good idea. Let's improve it afterwards.

**Review by swatDong — DISMISSED** [review]

**Review by swatDong — APPROVED** [review]

**Review by XiaofuHuang — APPROVED** [review]

**Review by hund030 — APPROVED** [review]

**Pull request merged**